### PR TITLE
Add ignore_docm option

### PR DESCRIPTION
**Pull request by dufeiyu**
For non-cle analysis, docm variants are not needed and this PR is to add ignore_docm option to docm_filter.pl to filter out all docm variants.

### CODE REVIEW

File: docm_filter.pl [hunk 1]
```diff
@@ -32,6 +32,9 @@
         say $docm_filter_fh $_;
     }
     elsif (/^#CHROM/) {
+        if ($ignore_docm_flag and $ignore_docm_flag eq 'ignoreDOCM') {
+            say $docm_filter_fh '##FILTER=<ID=IgnoreDOCM,Description="ignore Docm variants">';
```
Comment (chrisamiller) [inline docm_filter.pl]: From the perspective of someone looking at the FILTER field in the final VCF, I like DOCM_ONLY as the name of the filter (which answers the question "why was this filtered?")

**Review by chrisamiller — APPROVED** [review]
other than the renaming request, looks good!

File: docm_filter.pl (hunk 1 above)
Comment (dufeiyu) [reply docm_filter.pl]: yeah, it makes sense. Change is made.

**Comment by chrisamiller** [comment]
Requesting a re-review, since I pushed a minor change here

File: docm_filter.pl [hunk 2]
```diff
@@ -5,8 +5,8 @@
 
 use feature qw(say);
 
-die("Wrong number of arguments. Provide docm_out_vcf, normal_cram, tumor_cram, output_dir") unless scalar(@ARGV) == 4;
-my ($docm_out_vcf, $normal_cram, $tumor_cram, $outdir) = @ARGV;
+die("Wrong number of arguments. Provide docm_out_vcf, normal_cram, tumor_cram, output_vcf_file") unless scalar(@ARGV) == 4;
+my ($docm_out_vcf, $normal_cram, $tumor_cram, $outfile, $ignore_docm_flag) = @ARGV;
```
Comment (tmooney) [inline docm_filter.pl]: It's not possible for the `$ignore_docm_flag` to be set, since the previous line mandates only four parameters be specified.

**Comment by chrisamiller** [comment]
fixed that - thanks, Tom!

**Review by tmooney — APPROVED** [review]
+1, although merging this will immediately break existing workflows, since this line is not versioned:

https://github.com/genome/analysis-workflows/blob/5a4fed24f01f6d7c7ce1f595b272d82d97d4f9bd/definitions/tools/somatic_docm_filter.cwl#L9

**Comment by chrisamiller** [comment]
Not planning to merge it until the complete replacement is ready (that merges docm in the new way, etc).  That PR is still pending getting a run of the pipeline through on real data, since the chr17 subset doesn't contain any docm sites.  That has been a little more problematic than I'd like.

**Pull request merged**